components/service/nimbus/src/main/java/mozilla/components/service/nimbus/Nimbus.kt

```diff
@@ -53,14 +56,76 @@ interface NimbusApi : Observable<NimbusApi.Observer> {
      *
      * @return A String representing the branch-id or "slug"
      */
+    @AnyThread
```

Comment:
nit: In line with this, should we add `@WorkerThread` to all `*OnThisThread` methods on the interface?

Reply:
I've removed the `initializeOnThisThread` method from the interface, and made it test only.